### PR TITLE
Add support rms wallet

### PR DESCRIPTION
**Pull request by new4762**
Channel to support  

- Touch 'n Go
- GrabPay
- Boost
- ShopeePay
- Maybank QR Pay
- DuitNow QR & OBW
For Touch 'n Go & GrabPay we will add support provider for display correctly provider
In this case when provider is Alipay+ we will show `Alipay+ Partner` if not will show nothing

``Example for non RMS version for Grab and Touch 'n Go``

```json
{
	"object": "capability",
	"location": "/capability",
	"banks": [
		"9353",
		"9548"
	],
	"payment_methods": [
		{
			"object": "payment_method",
			"name": "boost",
			"currencies": [
				"MYR"
			],
			"card_brands": null,
			"installment_terms": null,
			"banks": [],
			"provider": null
		},
		{
			"object": "payment_method",
			"name": "grabpay",
			"currencies": [
			  "SGD",
				"MYR"
			],
			"card_brands": null,
			"installment_terms": null,
			"banks": [],
			"provider": null
		},
		{
			"object": "payment_method",
			"name": "shopeepay",
			"currencies": [
				"MYR"
			],
			"card_brands": null,
			"installment_terms": null,
			"banks": [],
			"provider": null
		},
		{
			"object": "payment_method",
			"name": "duitnow_obw",
			"currencies": [
				"MYR"
			],
			"card_brands": null,
			"installment_terms": null,
			"banks": [],
			"provider": null
		},
		{
			"object": "payment_method",
			"name": "duitnow_qr",
			"currencies": [
				"MYR"
			],
			"card_brands": null,
			"installment_terms": null,
			"banks": [],
			"provider": null
		},
		{
			"object": "payment_method",
			"name": "maybank_qr",
			"currencies": [
				"MYR"
			],
			"card_brands": null,
			"installment_terms": null,
			"banks": [],
			"provider": null
		},
		{
			"object": "payment_method",
			"name": "touch_n_go",
			"currencies": [
				"THB",
				"JPY",
				"SGD",
				"MYR"
			],
			"card_brands": null,
			"installment_terms": null,
			"banks": [],
			"provider": "Alipay_plus"
		}
	],
	"country": "MY",
	"tokenization_methods": [],
	"zero_interest_installments": false
}
```

https://user-images.githubusercontent.com/97080697/173328452-5a1966e2-c563-4352-8229-5b77a09e9522.mp4

``Example for RMS version ``

```json
{
	"object": "capability",
	"location": "/capability",
	"banks": [
		"9353",
		"9548"
	],
	"payment_methods": [
		{
			"object": "payment_method",
			"name": "boost",
			"currencies": [
				"MYR"
			],
			"card_brands": null,
			"installment_terms": null,
			"banks": [],
			"provider": null
		},
		{
			"object": "payment_method",
			"name": "grabpay",
			"currencies": [
			  "SGD",
				"MYR"
			],
			"card_brands": null,
			"installment_terms": null,
			"banks": [],
			"provider": "RMS"
		},
		{
			"object": "payment_method",
			"name": "shopeepay",
			"currencies": [
				"MYR"
			],
			"card_brands": null,
			"installment_terms": null,
			"banks": [],
			"provider": null
		},
		{
			"object": "payment_method",
			"name": "duitnow_obw",
			"currencies": [
				"MYR"
			],
			"card_brands": null,
			"installment_terms": null,
			"banks": [],
			"provider": null
		},
		{
			"object": "payment_method",
			"name": "duitnow_qr",
			"currencies": [
				"MYR"
			],
			"card_brands": null,
			"installment_terms": null,
			"banks": [],
			"provider": null
		},
		{
			"object": "payment_method",
			"name": "maybank_qr",
			"currencies": [
				"MYR"
			],
			"card_brands": null,
			"installment_terms": null,
			"banks": [],
			"provider": null
		},
		{
			"object": "payment_method",
			"name": "touch_n_go",
			"currencies": [
				"THB",
				"JPY",
				"SGD",
				"MYR"
			],
			"card_brands": null,
			"installment_terms": null,
			"banks": [],
			"provider": "RMS"
		}
	],
	"country": "MY",
	"tokenization_methods": [],
	"zero_interest_installments": false
}
```

https://user-images.githubusercontent.com/97080697/173328587-ab81036a-ee49-44ce-a6e3-99d99488f23c.mp4

### CODE REVIEW

File: sdk/src/sharedTest/java/co/omise/android/ui/PaymentChooserFragmentTest.kt [hunk 1]
```diff
@@ -236,7 +239,7 @@ class PaymentChooserFragmentTest {
         )
         verify(fragment.navigation)?.navigateToMobileBankingChooser(expectedMethods)
     }
-
+
```
Comment (hpariwat) [inline sdk/src/sharedTest/java/co/omise/android/ui/PaymentChooserFragmentTest.kt:242]: Whitespace

**Review by hpariwat — APPROVED** [review]

**Review by tookaa2 — APPROVED** [review]

File: sdk/src/sharedTest/java/co/omise/android/ui/PaymentChooserFragmentTest.kt (hunk 1 above)
Comment (new4762) [reply sdk/src/sharedTest/java/co/omise/android/ui/PaymentChooserFragmentTest.kt]: nice catch !! here removed -> https://github.com/omise/omise-android/pull/225/commits/fec5177185f6ffbfa23782949dedbb95f28202c6 🙏

**Comment by new4762** [comment]
> Channel to support
> 
>     * Touch 'n Go
> 
>     * GrabPay
> 
>     * Boost
> 
>     * ShopeePay
> 
>     * Maybank QR Pay
> 
>     * DuitNow QR & OBW
>       For Touch 'n Go & GrabPay we will add support provider for display correctly provider
>       In this case when provider is Alipay+ we will show `Alipay+ Partner` if not will show nothing
> 
> 
> `Example for non RMS version for Grab and Touch 'n Go`
> 
> ```json
> {
> 	"object": "capability",
> 	"location": "/capability",
> 	"banks": [
> 		"9353",
> 		"9548"
> 	],
> 	"payment_methods": [
> 		{
> 			"object": "payment_method",
> 			"name": "boost",
> 			"currencies": [
> 				"MYR"
> 			],
> 			"card_brands": null,
> 			"installment_terms": null,
> 			"banks": [],
> 			"provider": null
> 		},
> 		{
> 			"object": "payment_method",
> 			"name": "grabpay",
> 			"currencies": [
> 			  "SGD",
> 				"MYR"
> 			],
> 			"card_brands": null,
> 			"installment_terms": null,
> 			"banks": [],
> 			"provider": null
> 		},
> 		{
> 			"object": "payment_method",
> 			"name": "shopeepay",
> 			"currencies": [
> 				"MYR"
> 			],
> 			"card_brands": null,
> 			"installment_terms": null,
> 			"banks": [],
> 			"provider": null
> 		},
> 		{
> 			"object": "payment_method",
> 			"name": "duitnow_obw",
> 			"currencies": [
> 				"MYR"
> 			],
> 			"card_brands": null,
> 			"installment_terms": null,
> 			"banks": [],
> 			"provider": null
> 		},
> 		{
> 			"object": "payment_method",
> 			"name": "duitnow_qr",
> 			"currencies": [
> 				"MYR"
> 			],
> 			"card_brands": null,
> 			"installment_terms": null,
> 			"banks": [],
> 			"provider": null
> 		},
> 		{
> 			"object": "payment_method",
> 			"name": "maybank_qr",
> 			"currencies": [
> 				"MYR"
> 			],
> 			"card_brands": null,
> 			"installment_terms": null,
> 			"banks": [],
> 			"provider": null
> 		},
> 		{
> 			"object": "payment_method",
> 			"name": "touch_n_go",
> 			"currencies": [
> 				"THB",
> 				"JPY",
> 				"SGD",
> 				"MYR"
> 			],
> 			"card_brands": null,
> 			"installment_terms": null,
> 			"banks": [],
> 			"provider": "Alipay_plus"
> 		}
> 	],
> 	"country": "MY",
> 	"tokenization_methods": [],
> 	"zero_interest_installments": false
> }
> ```
> 
> non_rms_version.mp4
> 
> `Example for RMS version `
> 
> ```json
> {
> 	"object": "capability",
> 	"location": "/capability",
> 	"banks": [
> 		"9353",
> 		"9548"
> 	],
> 	"payment_methods": [
> 		{
> 			"object": "payment_method",
> 			"name": "boost",
> 			"currencies": [
> 				"MYR"
> 			],
> 			"card_brands": null,
> 			"installment_terms": null,
> 			"banks": [],
> 			"provider": null
> 		},
> 		{
> 			"object": "payment_method",
> 			"name": "grabpay",
> 			"currencies": [
> 			  "SGD",
> 				"MYR"
> 			],
> 			"card_brands": null,
> 			"installment_terms": null,
> 			"banks": [],
> 			"provider": "RMS"
> 		},
> 		{
> 			"object": "payment_method",
> 			"name": "shopeepay",
> 			"currencies": [
> 				"MYR"
> 			],
> 			"card_brands": null,
> 			"installment_terms": null,
> 			"banks": [],
> 			"provider": null
> 		},
> 		{
> 			"object": "payment_method",
> 			"name": "duitnow_obw",
> 			"currencies": [
> 				"MYR"
> 			],
> 			"card_brands": null,
> 			"installment_terms": null,
> 			"banks": [],
> 			"provider": null
> 		},
> 		{
> 			"object": "payment_method",
> 			"name": "duitnow_qr",
> 			"currencies": [
> 				"MYR"
> 			],
> 			"card_brands": null,
> 			"installment_terms": null,
> 			"banks": [],
> 			"provider": null
> 		},
> 		{
> 			"object": "payment_method",
> 			"name": "maybank_qr",
> 			"currencies": [
> 				"MYR"
> 			],
> 			"card_brands": null,
> 			"installment_terms": null,
> 			"banks": [],
> 			"provider": null
> 		},
> 		{
> 			"object": "payment_method",
> 			"name": "touch_n_go",
> 			"currencies": [
> 				"THB",
> 				"JPY",
> 				"SGD",
> 				"MYR"
> 			],
> 			"card_brands": null,
> 			"installment_terms": null,
> 			"banks": [],
> 			"provider": "RMS"
> 		}
> 	],
> 	"country": "MY",
> 	"tokenization_methods": [],
> 	"zero_interest_installments": false
> }
> ```
> 
> rms_version.mp4

FYI . This PR will merge after provider field is support on capability api

File: sdk/src/main/java/co/omise/android/models/SourceType.kt [hunk 2]
```diff
@@ -29,13 +29,22 @@ sealed class SourceType(
     object Dana : SourceType("dana")
     object Gcash : SourceType("gcash")
     object Kakaopay : SourceType("kakaopay")
-    object TouchNGo : SourceType("touch_n_go")
+    object TouchNGo : SourceType("touch_n_go") {
+        var provider : String? = null
+    }
```
Comment (nuxzero) [inline sdk/src/main/java/co/omise/android/models/SourceType.kt:34]:
```suggestion
    data class TouchNGo(val provider: String? = null) : SourceType("touch_n_go") 
```

File: sdk/src/main/java/co/omise/android/models/SourceType.kt [hunk 3]
```diff
@@ -29,13 +29,22 @@ sealed class SourceType(
...
     object OcbcPao : SourceType("mobile_banking_ocbc_pao")
-    object GrabPay : SourceType("grabpay")
+    object Boost: SourceType("boost")
+    object ShopeePay: SourceType("shopeepay")
+    object DuitNowOBW  : SourceType("duitnow_obw")
+    object DuitNowQR : SourceType("duitnow_qr")
+    object MaybankQR : SourceType("maybank_qr")
+    object GrabPay : SourceType("grabpay"){
+        var provider : String? = null
+    }
```
Comment (nuxzero) [inline sdk/src/main/java/co/omise/android/models/SourceType.kt:47]: Could you change to `data class` please?

Comment (new4762) [reply sdk/src/main/java/co/omise/android/models/SourceType.kt]: Updated here krub -> https://github.com/omise/omise-android/pull/225/commits/4fd9b21527ab952e5ec38956a94eb71c1109118b  🙏

**Review by nuxzero — APPROVED** [review]

**Pull request merged**